agents-api/agents_api/activities/task_steps/base_evaluate.py

```diff
@@ -67,41 +79,15 @@ async def base_evaluate(
 
     evaluator = get_evaluator(names=values, extra_functions=extra_lambdas)
 
-    chosen_expression = ""
-
     try:
-        result = None
-        match exprs:
-            case str():
-                chosen_expression = exprs
-                result = evaluator.eval(exprs)
-            case list():
-                result = []
-                for expr in exprs:
-                    chosen_expression = expr
-                    result.append(evaluator.eval(expr))
-            case dict() as d if all(
-                isinstance(v, dict) or isinstance(v, str) for v in d.values()
-            ):
-                result = {}
-                for k, v in d.items():
-                    if isinstance(v, str):
-                        chosen_expression = v
-                        result[k] = evaluator.eval(v)
-                    else:
-                        result[k] = {}
-                        for k1, v1 in v.items():
-                            chosen_expression = v1
-                            result[k][k1] = evaluator.eval(v1)
-            case _:
-                raise ValueError(f"Invalid expression: {exprs}")
-
+        # Replaced match-case logic with recursive evaluation
+        result = _recursive_evaluate(exprs, evaluator)
         return result
 
     except BaseException as e:
```

Comment:
This needs to move inside the recursive function call. Otherwise we won't know which actual string failed to evaluate